src/reactive_system_urc/test/reactive-system-test.cpp

```diff
@@ -0,0 +1,581 @@
+
+    EXPECT_FLOAT_EQ(traj[0].x, 0.5);
+    EXPECT_FLOAT_EQ(traj[0].y, 0);
+    EXPECT_FLOAT_EQ(traj[1].x, 1);
+    EXPECT_FLOAT_EQ(traj[1].y, 0);
+    EXPECT_FLOAT_EQ(traj[2].x, 1.5);
+    EXPECT_FLOAT_EQ(traj[2].y, 0);
+}
+
+TEST(ReactiveSystem, curvedTraj) {
+    float linear_vel  = 1;
+    float angular_vel = 1;
+    float time_inc    = 0.2;
+    int num_incs      = 5;
+
+    std::vector<sb_geom_msgs::Point2D> traj =
+    ReactiveSystemPath::getArcTrajectory(
+    linear_vel, angular_vel, time_inc, num_incs);
+
+    EXPECT_NEAR(traj[0].x, 0.199, 0.001);
+    EXPECT_NEAR(traj[0].y, 0.020, 0.001);
+    EXPECT_NEAR(traj[1].x, 0.389, 0.001);
+    EXPECT_NEAR(traj[1].y, 0.079, 0.001);
+    EXPECT_NEAR(traj[2].x, 0.565, 0.001);
+    EXPECT_NEAR(traj[2].y, 0.175, 0.001);
+    EXPECT_NEAR(traj[3].x, 0.717, 0.001);
+    EXPECT_NEAR(traj[3].y, 0.303, 0.001);
+    EXPECT_NEAR(traj[4].x, 0.841, 0.001);
+    EXPECT_NEAR(traj[4].y, 0.460, 0.001);
+}
+
+TEST(ReactiveSystem, curvedTrajOpposite) {
+    float linear_vel  = 1;
+    float angular_vel = -1;
+    float time_inc    = 0.2;
+    int num_incs      = 5;
+
+    std::vector<sb_geom_msgs::Point2D> traj =
+    ReactiveSystemPath::getArcTrajectory(
+    linear_vel, angular_vel, time_inc, num_incs);
+
+    EXPECT_NEAR(traj[0].x, 0.199, 0.001);
+    EXPECT_NEAR(traj[0].y, -0.020, 0.001);
+    EXPECT_NEAR(traj[1].x, 0.389, 0.001);
+    EXPECT_NEAR(traj[1].y, -0.079, 0.001);
+    EXPECT_NEAR(traj[2].x, 0.565, 0.001);
+    EXPECT_NEAR(traj[2].y, -0.175, 0.001);
+    EXPECT_NEAR(traj[3].x, 0.717, 0.001);
+    EXPECT_NEAR(traj[3].y, -0.303, 0.001);
+    EXPECT_NEAR(traj[4].x, 0.841, 0.001);
+    EXPECT_NEAR(traj[4].y, -0.460, 0.001);
+}
+
+TEST(ReactiveSystem, riskAreaBuilder) {
+    float risk_score  = 5;
+    float risk_radius = 1;
+    sb_geom_msgs::Point2D risk_center;
+    risk_center.x = 2;
+    risk_center.y = 0;
+
+    sb_geom_msgs::Point2D corner1;
+    corner1.x = 0;
+    corner1.y = -2;
+
+    sb_geom_msgs::Point2D corner2;
+    corner2.x = 4;
+    corner2.y = 2;
+
+    float risk_length = 0.05;
+    RiskAreaBuilder::RiskAreaBuilder builder(corner1, corner2, risk_length);
+
+    builder.addRiskZone(
+    risk_center, risk_radius, risk_score); // rad = 1, score = 5
+
+    mapping_msgs_urc::RiskAreaArray risks = builder.getRiskArray();
+
+    // For testing that riskareabuilder is correct
+    for (mapping_msgs_urc::RiskArea area : risks.areas) {
+        if (area.score.data == risk_score) { // confirm that risk scores were
+                                             // added in correct positions
+            EXPECT_NEAR(pow(area.area.points[0].x - risk_center.x, 2) +
+                        pow(area.area.points[0].y - risk_center.y, 2),
+                        0,
+                        pow(risk_radius, 2) + 0.15);
+        }
+    }
+}
+
+// Test the given twist, for a scenario where an object is directly between our
+// position and the goal
+TEST(ReactiveSystem, getArcPath1) {
```

Comment:
Good job on the tests! They are really comprehensive. Instead of calling this getArcPath1, getArcPath2, etc. The better way to name the tests would be "TEST(getArcPath, objectInDirectPath)". Do we have a convention for test naming @garethellis0 ?

Reply:
I've changed the names to be more descriptive for now but there is a convention I could change it again.